### PR TITLE
Add "Allow JavaScript" Option

### PR DESCRIPTION
**Pull request by GustavoKatel**
This adds a `allow_javascript` boolean option, which if true, does not sanitize the html before rendering in the template.

**Author notes and concerns**:
1. There's no integration tests yet. I think it'd be better to start the review before writing them. Also I'm doing some on-boarding on them.
2. I've added a couple of scenarios, but I'm not sure about them. Is it ok to leave them hardcoded?
3. Actions buttons were not working in the settings page. The bundled js was built to only include one set of actions buttons, which ended up working for the Editor page. Fix in 843fa2a02da45ea4e34515bff47408e3708abc9a
4. CodeMirror was introducing strange characters in the `tinymce` content due to https://github.com/christiaan/tinymce-codemirror/issues/26 . da155a77798c313ae5a3c92c37b5b131c8a046e8 is a workaround.

**Reviewers**
- [ ] @pkulkark 
- [ ] @bradenmacdonald

### CODE REVIEW

File: html_xblock/html.py [hunk 1]
```diff
@@ -28,6 +28,12 @@ class HTML5XBlock(StudioEditableXBlockMixin, XBlock):
         default=_('Text')
     )
     data = String(help=_('Html contents to display for this module'), default=u'', scope=Scope.content)
+    allow_javascript = Boolean(
+        display_name=_('Allow JavaScript execution'),
+        help=_('Whether js should be allowed or not in this module'),
+        default=False,
+        scope=Scope.settings
```
Comment (pkulkark) [inline html_xblock/html.py]: I think it makes more sense to have `scope=Scope.content` here as this field would be needed for all users and across all runs of the course. See https://github.com/open-craft/XBlock/blob/master/xblock/fields.py#L156.

Comment (GustavoKatel) [reply html_xblock/html.py]: You're right. Thanks @pkulkark :+1:

**Comment by GustavoKatel** [comment]
Thanks @pkulkark ! I've just added the doc strings.

**Comment by bradenmacdonald** [comment]
@GustavoKatel I'm experiencing two issues when I try to test this:

First, when I have the "Settings" tab open, I can't save any changes to the block - the Save button doesn't work. It only saves when I go back to the "Editor" tab and hit Save. This may be unrelated to your change but I'd appreciate if you can look at a fix. Is it happening for you?

Second, and more importantly, when I set "Allow JavaScript" to False, some invalid characters are showing up in the HTML:
screen shot 2018-11-26 at 10 32 35 am

When I set it to true, those disappear, so I think they're some problem with the sanitization.

**Comment by GustavoKatel** [comment]
> @GustavoKatel I'm experiencing two issues when I try to test this:
> 
> First, when I have the "Settings" tab open, I can't save any changes to the block - the Save button doesn't work. It only saves when I go back to the "Editor" tab and hit Save. This may be unrelated to your change but I'd appreciate if you can look at a fix. Is it happening for you?

Weird :thinking: Just checked `master` and it is happening indeed. I will take a look.

> Second, and more importantly, when I set "Allow JavaScript" to False, some invalid characters are showing up in the HTML:
> screen shot 2018-11-26 at 10 32 35 am
> 
> When I set it to true, those disappear, so I think they're some problem with the sanitization.

I see. You edited with the "Visual" (`tinymce`) editor right? I guess it is doing some sanitization by itself? I will take a look as well.

Thanks for testing this :D

**Comment by GustavoKatel** [comment]
I could reproduce and fix the actions buttons not working. The event listener was not including the buttons in the settings tab page. The change makes sure they are included. I'm looking into a fix for the special characters and I will update as soon as possible.

**Comment by bradenmacdonald** [comment]
Awesome, thanks! Let me know when it's ready for a final review.

**Comment by GustavoKatel** [comment]
`saveCursorPosition` in the `codemirror` plugin in `tinymce` was causing the invalid characters issue. There's an issue open for that: https://github.com/christiaan/tinymce-codemirror/issues/26

The fix disables `saveCursorPosition` and avoids it to insert the marker in the code.

Good to go :+1: 

Please let me know if there's anything else I can improve. Thanks a lot.

File: html_xblock/static/js/html.js [hunk 2]
```diff
@@ -211,7 +211,7 @@ function HTML5XBlock(runtime, element, data) {
     const content = (data.editor === "visual") ? tinymce.get("html5-textarea").getContent() : editor.getValue();
     const fields_data = getSettingsValues(fields);
     var errorMessage = "This may be happening because of an error with our server or your internet connection. Try refreshing the page or making sure you are online.";
-
+
```
[the remark this answers is not in the document]
Comment (GustavoKatel) [reply html_xblock/static/js/html.js]: Oh! I see. I will make sure it doesn't happen again.

**Comment by bradenmacdonald** [comment]
Thanks @GustavoKatel. Please go ahead and click "Squash and Merge".

**Pull request merged**